redbot/cogs/warnings/warnings.py

```diff
@@ -340,33 +340,57 @@ def __init__(self, bot: Red):
             pass
         await ctx.send(_("User {user} has been warned.").format(user=user))
 
-    @commands.command()
+    @commands.command(usage="")
     @commands.guild_only()
     async def warnings(
         self, ctx: commands.Context, user: Optional[Union[discord.Member, int]] = None
     ):
-        """List the warnings for the specified user.
-
-        Omit `<user>` to see your own warnings.
-
-        Note that showing warnings for users other than yourself requires
-        appropriate permissions.
-        """
+        """Check your own warnings."""
         if user is None:
             user = ctx.author
-        else:
+        else:  # TODO: remove this else statement in a release after 3.3.0 (will allow users time start using [p]warncheck)
             if not await is_admin_or_superior(self.bot, ctx.author):
                 return await ctx.send(
                     warning(_("You are not allowed to check warnings for other users!"))
                 )
+            else:
+                return await ctx.send(
+                    _("Checking other user's warnings has moved to `{prefix}warncheck`.").format(
+                        prefix=ctx.prefix
+                    )
+                )
 
-            try:
-                userid: int = user.id
-            except AttributeError:
-                userid: int = user
-                user = ctx.guild.get_member(userid)
-                user = user or namedtuple("Member", "id guild")(userid, ctx.guild)
+        msg = ""
+        member_settings = self.config.member(user)
+        async with member_settings.warnings() as user_warnings:
+            if not user_warnings.keys():  # no warnings for the user
+                await ctx.send(_("You have no warnings! Well done!"))
+            else:
+                for key in user_warnings.keys():
+                    mod = ctx.guild.get_member(user_warnings[key]["mod"])
+                    if mod is None:
+                        mod = discord.utils.get(
+                            self.bot.get_all_members(), id=user_warnings[key]["mod"]
+                        )
+                        if mod is None:
```

Comment:
Fetch_user should not be used here, since this command is usable but anyone, at this stage we should return a placeholder

Reply:
This is still relevant and should not use fetch_user